XDG_BASE_DIR support (linux, windows)

elkowar opened the pull request:
This PR adds XDG_BASE_DIR support on linux, and cleans up the path declarations slightly. Additionally, we move the embeddings and conversations directly to the SUPPORT_DIR on those platforms.
I _think_ that should also be done on MacOS in the future, but that has been left out here for now to not break existing users setups. 

Additionally, we move the SUPPORT_DIR into LocalAppData on windows for consistency.

fixes https://github.com/zed-industries/zed/issues/9308
fixes https://github.com/zed-industries/zed/issues/7155

Release Notes:

- N/A

elkowar committed:
Support XDG_BASE_DIR spec on linux

Co-authored-by: phisch <philipp.schaffrath@gmail.com>

elkowar committed:
Clean up and use more correct dirs on linux

Co-authored-by: phisch <philipp.schaffrath@gmail.com>

elkowar committed:
Clean up placement of embeddings and conversations

Only done on windows and linux to not break existing MacOS setups.
This should potentially also be done on MacOS in the future.

Co-authored-by: phisch <philipp.schaffrath@gmail.com>

elkowar commented:
@cla-bot check

mikayla-maki commented on crates/util/src/paths.rs:
```diff
@@ -12,46 +12,55 @@ lazy_static::lazy_static! {
     pub static ref DB_DIR: PathBuf = SUPPORT_DIR.join("db");
     pub static ref CRASHES_DIR: PathBuf = if cfg!(target_os = "macos") {
         HOME.join("Library/Logs/DiagnosticReports")
-    } else if cfg!(target_os = "windows") {
```
This variable is used for us to detect segfaults and other crashes which are recorded by the system but couldn't be intercepted by us. I don't know if it's possible for us to find this data on linux, but I expect there might be something on Windows.

mikayla-maki replied:
Perhaps we should make it an Option instead, that's only defined for platforms we can find this information on.

mikayla-maki commented on crates/util/src/paths.rs:
```diff
@@ -12,46 +12,55 @@ lazy_static::lazy_static! {
-        CONFIG_DIR.join("crashes")
+        SUPPORT_DIR.join("crashes")
     };
     pub static ref CRASHES_RETIRED_DIR: PathBuf = if cfg!(target_os = "macos") {
```
Same with this variable.

elkowar committed:
Make CRASHES_DIR and CRASHES_RETIRED_DIR Optional

These for now only make sense on MacOS, as no other platform
has a unified place for crash logs. Thus, we only set those paths when
on MacOS.
